backend-shared/config/db/apicrtodatabasemapping_test.go

```diff
@@ -57,4 +58,8 @@ func TestApiCRToDBMappingFunctions(t *testing.T) {
 	err = dbq.GetDatabaseMappingForAPICR(ctx, &fetchRow)
 	assert.True(t, IsResultNotFoundError(err))
 
+	// Set the invalid value
+	item.APIResourceName = strings.Repeat("abc", 100)
+	err = dbq.CreateAPICRToDatabaseMapping(ctx, &item)
+	assert.Error(t, err)
```

Comment:
For this `assert.Error()` (and others like it in your PR), I  recommend checking that the err string contains "value exceeds maximum size", to verify that your `validateFieldLength` logic is working as expected.

Something like:
```go
if err != nil {
 assert.True(t, strings.Contains(err.Error(), "value exceeds maximum size")
}
```